codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +277,13 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                int http1Index =
+                    ByteBufUtil.indexOf(HTTP_1_X_BUF, in.slice(in.readerIndex(), min(in.readableBytes(), 1024)));
```

Comment:
Added a comment.  There isn't a strong reason.  A number is needed so I picked one.  I think IE is around 2000 max path length, but that seems too long.  I'm content for the other error message to be returned in that case.

The RFC (7230, 7231) doesn't seem to specify a limit and implies it is up to the implementation.